src/main/scala/io/picnicml/doddlemodel/examples/preprocessing/BinarizerExample.scala

```diff
@@ -0,0 +1,26 @@
+package io.picnicml.doddlemodel.examples.preprocessing
+
+import breeze.linalg.DenseMatrix
+import io.picnicml.doddlemodel.data.Feature.{CategoricalFeature, FeatureIndex, NumericalFeature}
+import io.picnicml.doddlemodel.preprocessing.Binarizer
+import io.picnicml.doddlemodel.syntax.TransformerSyntax._
+
+object BinarizerExample extends App {
+  val x = DenseMatrix(
+    List(1.0, 1.0, 1.0),
+    List(3.0, 0.0, 1.5),
+    List(6.0, 2.0, 0.0)
+  )
+  val featureIndex = FeatureIndex(List(NumericalFeature, CategoricalFeature, NumericalFeature))
+  println(s"features: $featureIndex")
+
+  // binarize all numerical features
+  val binarizer = Binarizer(threshold = 1.0, featureIndex = featureIndex)
+  val fittedBinarizer = binarizer.fit(x)
+  println(s"binarized data:\n${fittedBinarizer.transform(x)}")
+
+  // only binarize the last feature
+  val binarizerSubset = Binarizer(threshold = 1.0, featureIndex = featureIndex.subset("f2"))
```

Comment:
This is just an idea, not really a complaint: should we avoid using `.subset(<String*>)` on feature indices that do not have explicitly named attributes and instead either explicitly name attributes or use the method with an `Int` sequence?
 
That way, in the (probably unlikely) case that the naming scheme for features is changed, we do not have to fix these examples.

Another option I see is to just mention how attributes are named in the docs (I can do that in the already opened PR)

Reply:
You are right, the use of auto-generated names assumes implicit knowledge, I'll go ahead and change that. Also, having it in docs is a good idea.

Reply:
A better API of that function would also be beneficial, e.g. `subset(0, 1)` instead of `subset(IndexedSequence(0, 1))`.